substrate/frame/staking/rc-client/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+//! before redirecting them to the staking interface.
+//!
+//! On the opposite direction, **outbout** requests from staking to external consensus systems are
+//! implemented by the [`crate::AsyncBroker`] trait. The implementor is responsible to pre-process
+//! requests and, potentially, forward the requests to an external consensus system (e.g. relay
+//! chain).
+//!
+//! One of the design goals of the [`crate::AsyncBroker`] trait and implementation is to abstract
+//! all the complexity derived from sync/async and the fact that the consumers and producers of the
+//! data expected by staking *may* be in a different consensus system. From the staking implementor
+//! POV, it should be transparent whether the broker is between e.g. the session manager or not.
+//!
+//! ## Async vs Sync communication
+//!
+//! This pallet is designed to support an async comminication channel between staking and the
+//! session interfaces. This pallet should abstract the potential asynchronicity of the
+//! channel to the staking interface. As such and when possible, this pallet caches information
+//! from external consensus systems so that the requests from staking can be served in a
+//! synchronous manner. For example, this pallet caches the current validator set as seen by the
+//! external consensus system to be able to serve the from staking request without having to query
+//! the external consensus system, which is ultimately the source of truth for the active validator
+//! set.
+//!
+//! ## Inbound flow
+//!
+//! All the inbound request should be performed through extrinsics. External consensus systems may
+//! call the inbound extrinsics through XCM Transact.
+//!
+//! ### Block authoring
+//!
+//! This pallet exposes an extrinsict, [`Call::author`], that processes block authoring events.
+//! Block authoring information can only be submitted by the runtime's root origin. Successfull
+//! calls will be redirected to [`Config::Staking`] through the [`pallet_authorship::EventHandler`])
+//! interface.
+//!
+//! ### Offence reports
+//!
+//! This pallet exposes an extrinsict, [`Call::report_offence`], that processes priviledged offence
+//! reports. These reports can only be submitted by the runtime's root origin. Successfull calls
+//! will be redirected to staking through the [`sp_staking::offence::OnOffenceHandler`]) interface.
+//!
+//! ## Outbound flow
+//!
+//!	### Validator keys
+//!
+//!	This pallet implements a set of extrinsics that handles session key management requests for
+//! staking validators. Note, however, that this pallet is not the source of truth for session
+//! keys. It only exposes interfaces for accounts to request session key initialization and
+//! termination, performs pre-checks and propagates that information to another pallet or consensus
+//! system through the [`crate::Config::SessionKeysHandler`].
+//!
+//!	Callers can request to 1) set validator keys and 2) purge validator keys. These actions *may
```

Comment:
Is this any different from calling `pallet_xcm::send_xcm(transact { relay_chain_call })`? I suggest consulting with @franciscoaguirre in the high level of design of RC <> AH  interaction.